usso.go

```diff
@@ -95,6 +98,47 @@ func (server UbuntuSSOServer) GetTokenWithOTP(email, password, otp, tokenName st
 	return &ssodata, nil
 }
 
+// Error represents an error message returned from Ubuntu SSO.
+type Error struct {
+	Message string                 `json:"message"`
+	Code    string                 `json:"code,omitempty"`
+	Extra   map[string]interface{} `json:"extra,omitempty"`
+}
+
+// getError attempts to extract the most meaningfule error that it can
+// from a response.
+func getError(resp *http.Response) *Error {
+	var ssoError Error
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		ssoError.Code = resp.Status
+		ssoError.Message = resp.Status
+		return &ssoError
+	}
+	err = json.Unmarshal(body, &ssoError)
+	if err != nil {
+		// Attempt to pass the original error back in the best way possible
+		ssoError.Code = resp.Status
+		ssoError.Message = string(body)
+		return &ssoError
+	}
+	return &ssoError
```

Comment:
They won't because they're not of the correct type Tests are a good idea anyway, so I've added some.